scripts/run_inference.py

```diff
@@ -183,13 +183,30 @@ def load(
                 try:
                     if base_model_name_or_path is None:
                         raise ValueError("base_model_name_or_path has to be passed")
-                    base_model = AutoModelForCausalLM.from_pretrained(
-                        base_model_name_or_path,
-                        attn_implementation="flash_attention_2"
-                        if use_flash_attn
-                        else None,
-                        torch_dtype=torch.bfloat16 if use_flash_attn else None,
+
+                    is_quantized = os.path.exists(
+                        os.path.join(base_model_name_or_path, "quantize_config.json")
                     )
+                    if is_quantized:
```

Comment:
Would be interested to know if these are common across other quantized models and if the configurations set below work for other quantize tuning techniques.

Reply:
This config here is specific only to GPTQ for the faster exllamav2 kernel. HF's `from_pretrained` will accept a quantization_config from any of [these configs,](https://huggingface.co/docs/transformers/en/main_classes/quantization) so it might be better to generalize this in future.

Reply:
@achew010 If they do not use anything else other than `gptq` then this is fine.